tools/scoring/score/dimensions/dimension.py

```diff
@@ -176,8 +180,123 @@ def is_entity_virtual(*, entity: EntityInstance) -> bool:
     """
     return entity.links is not None
 
+  @staticmethod
+  def match_virtual_entities(
+      *, solution_points_virtual: List[Tuple[Set[str], EntityType]],
```

Comment:
what is the key here? cloud_device_id?

Reply:
do you want to create a structure?
`List[fooStrcture]`?

Reply:
> what is the key here? cloud_device_id?

Are you asking about `str`? They are raw field names.

> do you want to create a structure?
> `List[fooStrcture]`?

Possibly a good idea — will consider implementation.